src/PrestaShopBundle/Resources/config/services/adapter/product/_combination.yml

```
@@ -57,6 +57,8 @@ services:
             - '@prestashop.adapter.product.stock.repository.stock_available_repository'
             - '@prestashop.core.grid.query_builder.product_combination'
             - '@prestashop.adapter.attribute.repository.attribute_repository'
+            - '@prestashop.adapter.product.image.repository.product_image_repository'
+            - '@prestashop.adapter.product.image.product_image_path_factory'
```

```suggestion
            - '@prestashop.adapter.product.image.product_image_url_factory'
```

The query handler returns the URLs, so it should use this service